chain/network/src/peer_manager.rs

```diff
@@ -681,16 +685,36 @@ impl PeerManagerActor {
     }
 }
 
+// TODO Incoming needs someone to own TcpListener, temporary workaround until there is a better way
+pub struct IncomingCrutch {
+    listener: TcpListener,
+}
+
+impl Stream for IncomingCrutch {
+    type Item = std::io::Result<TcpStream>;
+
+    fn poll_next(
+        mut self: Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> Poll<Option<Self::Item>> {
+        Stream::poll_next(std::pin::Pin::new(&mut self.listener.incoming()), cx)
+    }
+}
+
 impl Actor for PeerManagerActor {
     type Context = Context<Self>;
 
     fn started(&mut self, ctx: &mut Self::Context) {
         // Start server if address provided.
         if let Some(server_addr) = self.config.addr {
             // TODO: for now crashes if server didn't start.
-            let listener = TcpListener::bind(&server_addr).unwrap();
+            let listener = std::net::TcpListener::bind(&server_addr).unwrap();
+            let listener = TcpListener::from_std(listener).unwrap();
+            let incoming = IncomingCrutch { listener };
```

Comment:
@frol is there a better way to do this?

Reply:
@bowenwang1996 There is not something that I can suggest right now. This does not look terrible to me.